core/modules/wpublish/QueriesAndChunks.cc

```diff
@@ -219,6 +219,8 @@ void QueriesAndChunks::removeDead() {
     for (auto const& dead : dList) {
         removeDead(dead);
     }
+    lock_guard<mutex> gdend(_deadMtx);
```

Comment:
What's the purpose of the lock? Could it be commented on here?

Reply:
It just protects _deadQueries. I put the lock and the log line in debug log lvl check so it only locks when it is logging the line.

Reply:
Should the lock be acquired before iterating over `dList` and calling `removeDead`?

Reply:
removeDead(list) operates on different lists, locking the appropriate mutexes as it needs.